src/app/services/utility-services/user-info.service.ts

```diff
@@ -1,23 +1,25 @@
 import { Injectable } from '@angular/core';
+import { BehaviorSubject } from 'rxjs';
+
 import { IUserInfo } from 'src/app/login/login.model';
 
 @Injectable({
   providedIn: 'root'
 })
 export class UserInfoService {
-  private userInfo: IUserInfo = {
+  private userInfo$ = new BehaviorSubject<IUserInfo>({
```

Comment:
Any specific reason for using `subjects` ?

Reply:
I realised that it's not necessary that somebody requests for user info only after it's available. I had used this to implement showing the user `photoUrl` in the header, and didn't have a way to "listen" to changes. I used the `ngx-auth-firebaseui-avatar` instead, but left this as I feel it's a good way to expose the user information.

Reply:
If a simple service can do the work, no need to use subject. Unnecessary complication.